src/comments.js

```diff
@@ -449,19 +450,20 @@ function printDanglingComments(path, options, noIndent) {
     commentPath => {
       const comment = commentPath.getValue();
       if (!comment.leading && !comment.trailing) {
-        if (util.hasNewline(text, locStart(comment), { backwards: true })) {
-          parts.push(hardline);
-        }
         parts.push(printComment(commentPath));
       }
     },
     "comments"
   );
 
+  if (parts.length === 0) {
+    return "";
+  }
+
   if (!noIndent) {
-    return indent(options.tabWidth, concat(parts));
+    return indent(options.tabWidth, concat([hardline, join(hardline, parts)]));
```

Comment:
`noIndent` now has a lot more meaning, and the name is incorrect. Now `noIndent` means whether or not to force the comment on a new line or not. Please change the name of the option to something else to indicate this.